drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -30,6 +36,96 @@ func (k *k8sSchedOps) EnableOnNode(n node.Node) error {
 	return k8sops.Instance().RemoveLabelOnNode(n.Name, k8sPxRunningLabelKey)
 }
 
+func (k *k8sSchedOps) ValidateVolumeCleanup(sched scheduler.Driver, d node.Driver) error {
+	nodeToPodsMap := make(map[string][]string)
+	nodeMap := make(map[string]node.Node)
+
+	connOpts := node.ConnectionOpts{
+		Timeout:         1 * time.Minute,
+		TimeBeforeRetry: 10 * time.Second,
+	}
+	listVolOpts := node.FindOpts{
+		ConnectionOpts: connOpts,
+		Name:           "*portworx-volume",
+	}
+
+	for _, n := range sched.GetNodes() {
+		if n.Type == node.TypeWorker {
+			volDirList, _ := d.FindFiles(k8sPodsRootDir, n, listVolOpts)
+			nodeToPodsMap[n.Name] = separateFilePaths(volDirList)
+			nodeMap[n.Name] = n
+		}
+	}
+
+	existingPods, _ := k8sops.Instance().GetPods("")
+
+	orphanPodsMap := make(map[string][]string)
+	dirtyVolPodsMap := make(map[string][]string)
+
+	for nodeName, volDirPaths := range nodeToPodsMap {
+		var orphanPods []string
+		var dirtyVolPods []string
+
+		for _, path := range volDirPaths {
+			podUID := extractPodUID(path)
+			found := false
+			for _, existingPod := range existingPods.Items {
+				if podUID == string(existingPod.UID) {
+					found = true
+					break
+				}
+			}
+			if found {
+				continue
+			}
+			orphanPods = append(orphanPods, podUID)
+
+			// check if there are files under portworx volume
+			n := nodeMap[nodeName]
+			findFileOpts := node.FindOpts{
+				ConnectionOpts: connOpts,
```

Comment:
Can you add an comment showing example why depth of 2 is selected here with the path regex *portworx-volume?

Reply:
Done